pkg/tasks/nodes.go

```diff
@@ -131,6 +132,21 @@ func labelNodeOSes(s *state.State) error {
 
 			node.Labels["v1.kubeone.io/operating-system"] = string(host.OperatingSystem)
 
+			for labKey, labVal := range host.Labels {
+				if strings.HasSuffix(labKey, "-") {
+					// additional `if` instead of `if strings.HasSuffix(labKey, "-") && labVal == ""` is used to avoid
+					// entering `else` section when the labVal is not empty.
+					if labVal == "" {
```

Comment:
I see that if a label has `-` suffix and a value, we ignore it. I think we should have some API validation to reject such cases.

Reply:
Good point, fixed